addon/components/file-browser/component.js

```diff
@@ -2,155 +2,344 @@ import Ember from 'ember';
+                    }
+                }
+                item.set('isSelected', true);
             }
+            Ember.run.next(this, function(){
+                if (this.get('selectedItems.length') === 1) {
+                    this.set('shiftAnchor', item)
+                }
+            });
         },
-
-        navigateUp() {
-            let breadcrumbs = this.get('breadcrumbs');
-            if (breadcrumbs.length === 1) {
-                return;
+        viewItem() {
+            let item = this.get('selectedItems.firstObject');
+            this.sendAction('openFile', item);
+        },
+        openItem(item, qparams) {
+            this.sendAction('openFile', item, qparams);
+        },
+        downloadItem() {
+            let downloadLink = this.get('selectedItems.firstObject.links.download');
+            window.location = downloadLink;
+        },
+        _deleteItem(item, url) {
+            authenticatedAJAX({
+                url: url,
+                type: 'DELETE',
+                xhrFields: {withCredentials: true}
+            })
+            .done(() => {
+                //TODO rethink the flash system, this seems gross nooddly
+                item.set('flash', {
+                    message: 'This file has been deleted',
+                    type: 'danger'
+                });
+                setTimeout(() => {
+                    this.get('_items').removeObject(item);
+                    this.notifyPropertyChange('_items');
+                }, 1800);
+            })
+            .fail(() => {
+                item.set('flash', {
+                    message: 'Delete failed',
+                    type: 'danger'
+                });
+            });
+        },
+        deleteItem(){
+            let item = this.get('selectedItems.firstObject');
+            let url = item.get('links.download');
+            this.send('_deleteItem', item, url);
+            this.set('modalOpen', false);
+        },
+        deleteItems() {
+            for (let item_ of this.get('selectedItems')) {
+                let url = item_.get('links.download');
+                this.send('_deleteItem', item_, url);
+            }
+            this.set('modalOpen', false);
+        },
+        _rename(conflict) {
+            let item = this.get('selectedItems.firstObject');
+            this.set('modalOpen', false);
+            authenticatedAJAX({
+                url: item.get('links.upload'),
+                type: 'POST',
+                xhrFields: {withCredentials: true},
+                headers: {
+                    'Content-Type': 'Application/json'
+                },
+                data: JSON.stringify({
+                    action: 'rename',
+                    rename: this.get('textValue'),
+                    conflict: conflict || 'replace'
+                })
+            })
```

Comment:
This might be a little trickier to push into the model layer, but still probably doable.